src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -65,6 +65,25 @@ typedef struct {
   tsi_handshaker_result *handshaker_result;
 } security_handshaker;
 
+static size_t move_read_buffer_into_handshake_buffer(grpc_exec_ctx *exec_ctx,
+                                                     security_handshaker *h) {
+  if (h->handshake_buffer_size < h->args->read_buffer->length) {
+    h->handshake_buffer = (uint8_t *)gpr_realloc(h->handshake_buffer,
+                                                 h->args->read_buffer->length);
+    h->handshake_buffer_size = h->args->read_buffer->length;
+  }
+  size_t offset = 0;
+  size_t bytes_in_read_buffer = h->args->read_buffer->length;
```

Comment:
Since we're defining this anyway, might as well move it back to the top of the function and use it instead of `h->args->read_buffer->length` on lines 70, 72, and 73.